Newsletter for March 2026 (#32589)
idaholab/moose: modules/doc/content/newsletter/2026/2026_03.md

The [SubdomainsFromPartitionerGenerator.md] was added in the same effort to be able to test out various options for
assigning subdomains on the surface mesh before re-triangulating it.

### Constructive Solid Geometry support improvements

The `CSG` support was increased to:

- handle transformations of cells, universes, lattices and surfaces. Transformation supported include rotations, translations and scaling.
- provide 'delete' and 'swap' operations for changing the content of cells.

### Sampler Vectorpostprocessor sorting

The `SamplerBase` class used by sampler vectorpostprocessors has been expanded to accept a sampled variable as the `sort_by`
quantity. This lets the user specify an arbitrary sampling order for the sampled rows, as long as it can be expressed in an
auxiliary variable.

### Layered user objects improvements

The bounds of the subdivisions of layers in the `LayeredBase`-derived can now be specified individually rather than with
a blanket uniform-per-layer parameter. This aligns with the flexibility offered by the mesh extruder generators, where the layers
can be specified individually.

> ### 1D Spline utilities

Reviewer: ```suggestion
### 1D spline utilities
```

A B-Spline class and an associated `SplineUtils` class were added to MOOSE to be able to represent and sample a uniform clamped B-spline.
These splines offer the advantage that they can pass through a determined start and end point with an imposed starting and ending direction,
avoiding slope "breaks" at the connection with other 1D line meshes.
A `BSplineMeshGenerator` should be added to MOOSE very soon for such 1D line meshes.

### Miscellaneous String output improvements

`moose::stringify` was extended to support `std::tuple` of arbitrary sizes. The delimiter specified is used for the tuple, but not for
any container contained within the tuple.

`std::unordered_map` and `std::map` will now be output with `mooseUtils::prettyCppType` without the additional C++ type information such
as the `std::allocator`. This improves the readability of messages referring to map types.

### CSVDiff testing enhancement

> The CSVDiff testing utility now outputs differences for every time step, similarly to how the Exodiff reports differences between the current

Reviewer: ```suggestion
The CSVDiff testing utility now outputs differences for every time step, similarly to how the Exodiff tests report differences between the current
```

Several postprocessors, notably to output the residual or the number of solver iterations, can now report these metrics with the
[Eigenvalue.md] executioner.

### Block deletion generator usability improvement

The [BlockDeletionGenerator.md] can now accept a 'keep' operation so that only the list of blocks that should not be deleted can be specified.
This is shorter to specify if most of the blocks in a mesh should be deleted.

### MFEM Backend improvements

- Adaptive mesh refinement functionality is now available for static problems. Though both
  h- and p-refinement implementations are available, the user cannot perform hp-refinement
  or compound multiple iterations of p-refinement. This is due to challenges deciding whether
  to use h- or p-refinement and in renewing error estimates after a p-refinement iteration.
  ([idaholab/moose#30943](https://github.com/idaholab/moose/pull/30943))
- Material coefficients are now boundary restrictable too. This is to allow boundary
  integrators to use coefficients defined on the boundary itself.
  ([idaholab/moose#32541](https://github.com/idaholab/moose/pull/32541))

> - Fixed a couple of bugs: when converting names to attribute ids, we no longer assume their

Reviewer: ```suggestion
- A couple of bugs were fixed: when converting names to attribute ids, we no longer assume their
```
